packages/internal-scripts/src/add-browser-exports.ts

```diff
@@ -0,0 +1,213 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// This script exists to perform a one time operation of adding the "browser"
+// exports field to relevant package.json files. Motivated by:
+// https://github.com/lit/lit/issues/4446
+//
+// The script is idempotent.
+//
+// # Instructions
+// After building the script run:
+// `node packages/internal-scripts/bin/add-browser-exports.js`
+
+import assert from 'assert';
+import {execSync} from 'child_process';
+import {readFile, writeFile} from 'fs/promises';
+import {resolve} from 'path';
+
+testScriptIntegrity();
+
+/**
+ * A loose structure for what we might find in the "exports" value within a
+ * `package.json` file.
+ */
+interface Exports {
+  [path: string]: {
+    types?: string;
+    browser?: {
+      development?: string;
+      default?: string;
+    };
+    node?: {
+      development?: string;
+      default?: string;
+    };
+    development?: string;
+    default?: string;
+  };
+}
+
+const result = execSync(`find . -name package.json`);
+
+const packageJsonFiles = result
+  .toString()
+  .trim()
+  .split('\n')
+  .filter((filePath) => !filePath.includes('/node_modules/'))
+  .filter((filePath) => !filePath.includes('/test-files/'))
+  .filter((filePath) => !filePath.includes('/test-goldens/'))
+  .filter((filePath) => !filePath.includes('/goldens/'));
+
+const absolutePackageJsonFilePaths = packageJsonFiles.map((relativePath) =>
+  resolve(process.cwd(), relativePath)
+);
+
+export const run = async () => {
+  const work: Promise<void>[] = [];
+  for (const packageJsonPath of absolutePackageJsonFilePaths) {
+    work.push(
+      (async (): Promise<void> => {
+        const parsedPackageJson = JSON.parse(
+          await readFile(packageJsonPath, {encoding: 'utf8'})
+        );
+        const exports: Exports | string | undefined =
+          parsedPackageJson['exports'];
+        if (exports === undefined) {
+          return;
+        }
+        // `"exports": "./index.js"` can be skipped.
+        if (typeof exports === 'string') {
+          return;
+        }
+        const fixedExports = optionallyAddBrowserExportCondition(exports);
+        parsedPackageJson['exports'] = fixedExports;
+        await writeFile(
```

Comment:
Consider doing nothing if the exports are the same before and after (maybe optionallyAddBrowserExportCondition could communnicate whether it made any changes as part of its return), and if something does need to be changed, writing it to the console like:

```
Updated packages/lit-html/package.json
```

Reply:
Love it! Can do!

Reply:
Done!